pkg/volume/csi/csi_block.go

```diff
@@ -14,6 +14,55 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+/*
+This file defines block volume related methods for CSI driver.
+CSI driver is responsible for staging/publishing volumes to their staging/publish paths.
+Mapping and unmapping of a device in a publish path to its global map path and its
+pod device map path are done by operation_executor through MapBlockVolume/UnmapBlockVolume
+(MapBlockVolume and UnmapBlockVolume take care for lock, symlink, and bind mount).
+
+Summary of block volume related CSI driver's methods are as follows:
+ - GetGlobalMapPath returns a global map path,
+ - GetPodDeviceMapPath returns a pod device map path and filename,
+ - SetUpDevice calls CSI's NodeStageVolume and stage a volume to its staging path,
+ - MapPodDevice calls CSI's NodePublishVolume and publish a volume to its publish path,
+ - UnmapPodDevice calls CSI's NodeUnpublishVolume and unpublish a volume from its publish path,
+ - TearDownDevice calls CSI's NodeUnstageVolume and unstage a volume from its staging path.
+
+These methods are called by below sequences:
+ operation_executor    pkg/volume/csi     pkg/volume/util          ASW                 CSI (SP)
+ ------------------ -------------------- ----------------- ---------------------- ------------------
+   MountVolume
+          +--------> GetGlobalMapPath
+          +--------> SetupDevice ------------------------------------------------> NodeStageVolume
+          +-----------------------------------------------> MarkDeviceAsMounted
```

Comment:
do we need this many dashes? It makes it hard to read on smaller screens

Reply:
in general for this and the below "After successful MountVolume..." section I think it's better to not rely on formatting based on spacing as it can be difficult to read and may differ depending on dev configuration. Looks like this could probably just be simplified into a nested bulleted list type structure with the occasional arrow